src/crypto/PublicKey.ts

```diff
@@ -211,6 +217,123 @@ export class SignedPublicKey
   }
 }
 
+export class AccountLinkedPublicKeyV1
+  extends UnsignedPublicKey
+  implements publicKey.AccountLinkedPublicKey_V1
+{
+  keyBytes: Uint8Array
+  staticSignature: AccountLinkedStaticSignature | undefined
+  siweSignature: signature.AccountLinkedSIWESignature | undefined
+
+  constructor(obj: publicKey.AccountLinkedPublicKey_V1) {
+    if (!obj.keyBytes) {
+      throw new Error('Invalid AccountLinkedPublicKeyV1 has no keyBytes')
+    }
+    super(publicKey.UnsignedPublicKey.decode(obj.keyBytes))
+    this.keyBytes = obj.keyBytes
+    if (obj.staticSignature) {
+      this.staticSignature = new AccountLinkedStaticSignature(
+        obj.staticSignature
+      )
+    } else {
+      throw new Error(
+        'Unimplemented signature type for AccountLinkedPublicKeyV1'
+      )
+    }
+  }
+
+  // Return bytes of the encoded unsigned key.
+  private bytesToSign(): Uint8Array {
+    return this.keyBytes
+  }
+
+  public getLinkedAddress(role: AccountLinkedRole): string {
```

Comment:
Like seeing the accountLinkedRole System!

I initially assumed validation would occur on initialization, but I can see why coupling it with getAddress makes sense and keeps the number of types under control.